src/translator/service.cpp

```diff
@@ -6,33 +6,46 @@
 #include <utility>
 
 inline std::vector<marian::Ptr<const marian::Vocab>>
-loadVocabularies(marian::Ptr<marian::Options> options) {
+loadVocabularies(marian::Ptr<marian::Options> options, const std::vector<marian::string_view> *vocabsMemoryView) {
   // @TODO: parallelize vocab loading for faster startup
-  auto vfiles = options->get<std::vector<std::string>>("vocabs");
-  // with the current setup, we need at least two vocabs: src and trg
-  ABORT_IF(vfiles.size() < 2, "Insufficient number of vocabularies.");
-  std::vector<marian::Ptr<marian::Vocab const>> vocabs(vfiles.size());
-  std::unordered_map<std::string, marian::Ptr<marian::Vocab>> vmap;
-  for (size_t i = 0; i < vocabs.size(); ++i) {
-    auto m =
-        vmap.emplace(std::make_pair(vfiles[i], marian::Ptr<marian::Vocab>()));
-    if (m.second) { // new: load the vocab
-      m.first->second = marian::New<marian::Vocab>(options, i);
-      m.first->second->load(vfiles[i]);
+  std::vector<marian::Ptr<marian::Vocab const>> vocabs;
+  if(vocabsMemoryView != nullptr){
+    // load vocabs from buffer
+    vocabs.resize(vocabsMemoryView->size());
+    for (size_t i = 0; i < vocabs.size(); i++) {
+      marian::Ptr<marian::Vocab> vocab = marian::New<marian::Vocab>(options, i);
+      vocab->loadFromSerialized((*vocabsMemoryView)[i]);
```

Comment:
convert AlignedMemory -> string_view here somehow? You won't have to copy the underlying memory I'm guessing? Restrict this conversion troubles to minimum interaction with sentencepiece (I'm guessing method signature there is what leads to troubles). string_view will be alive while AlignedMemory is alive.

As with others, we accept `AlignedMemory` and `Service` owns `AlignedMemory` (for vocab).